ParseFile panic:.git/logs/refs/heads/main.go

**Describe the bug**
Branch names ending in `.go` crash `swag init`, either local or remote. 

Full error:
```
$ swag init
2018/05/14 18:06:42 Generate swagger docs....
2018/05/14 18:06:42 Generate general API Info
2018/05/14 18:06:42 ParseFile panic:.git/logs/refs/heads/main.go:1:1: expected 'package', found 'INT' 0000000000000000000000000000000000000000 (and 1 more errors)
panic: ParseFile panic:.git/logs/refs/heads/main.go:1:1: expected 'package', found 'INT' 0000000000000000000000000000000000000000 (and 1 more errors)

goroutine 1 [running]:
log.Panicf(0x8115e4, 0x13, 0xc4201610f8, 0x1, 0x1)
	/usr/local/go/src/log/log.go:333 +0xda
github.com/swaggo/swag.(*Parser).getAllGoFileInfo.func1(0xc4204ae160, 0x1c, 0x85af20, 0xc4204b08f0, 0x0, 0x0, 0x7, 0xc4201611a0)
	/redacted/go/src/github.com/swaggo/swag/parser.go:544 +0x1a3
path/filepath.walk(0xc4204ae160, 0x1c, 0x85af20, 0xc4204b08f0, 0xc420439550, 0x0, 0x0)
	/usr/local/go/src/path/filepath/path.go:357 +0x402
path/filepath.walk(0xc4204ae100, 0x14, 0x85af20, 0xc4204b0820, 0xc420439550, 0x0, 0x0)
	/usr/local/go/src/path/filepath/path.go:381 +0x2c2
path/filepath.walk(0xc4204b40c0, 0xe, 0x85af20, 0xc4204b0750, 0xc420439550, 0x0, 0x0)
	/usr/local/go/src/path/filepath/path.go:381 +0x2c2
path/filepath.walk(0xc4204b4060, 0x9, 0x85af20, 0xc4204b05b0, 0xc420439550, 0x0, 0x0)
	/usr/local/go/src/path/filepath/path.go:381 +0x2c2
path/filepath.walk(0xc42043a230, 0x4, 0x85af20, 0xc420497040, 0xc420439550, 0x0, 0x0)
	/usr/local/go/src/path/filepath/path.go:381 +0x2c2
path/filepath.walk(0x80c9d7, 0x2, 0x85af20, 0xc420496f70, 0xc420439550, 0x0, 0x10)
	/usr/local/go/src/path/filepath/path.go:381 +0x2c2
path/filepath.Walk(0x80c9d7, 0x2, 0xc420439550, 0x0, 0x0)
	/usr/local/go/src/path/filepath/path.go:403 +0x106
github.com/swaggo/swag.(*Parser).getAllGoFileInfo(0xc4203fd720, 0x80c9d7, 0x2)
	/redacted/go/src/github.com/swaggo/swag/parser.go:537 +0x70
github.com/swaggo/swag.(*Parser).ParseAPI(0xc4203fd720, 0x80c9d7, 0x2, 0x80da06, 0x7)
	/redacted/go/src/github.com/swaggo/swag/parser.go:63 +0x9d
github.com/swaggo/swag/gen.(*Gen).Build(0xc420161a28, 0x80c9d7, 0x2, 0x80da06, 0x7, 0x80f9e6, 0xe, 0x0, 0x0, 0x0, ...)
	/redacted/go/src/github.com/swaggo/swag/gen/gen.go:28 +0x287
main.main.func1(0xc42008c160, 0x0, 0xf3)
	/redacted/go/src/github.com/swaggo/swag/cmd/swag/main.go:23 +0x108
github.com/urfave/cli.HandleAction(0x788d40, 0x825fd8, 0xc42008c160, 0x0, 0xc420492780)
	/redacted/go/src/github.com/urfave/cli/app.go:501 +0xc8
github.com/urfave/cli.Command.Run(0x80cf63, 0x4, 0x0, 0x0, 0xc4204393c0, 0x1, 0x1, 0x80fa8e, 0xe, 0x0, ...)
	/redacted/go/src/github.com/urfave/cli/command.go:165 +0x47d
github.com/urfave/cli.(*App).Run(0xc420164000, 0xc42000c080, 0x2, 0x2, 0x0, 0x0)
	/redacted/go/src/github.com/urfave/cli/app.go:259 +0x6e8
main.main()
	/redacted/go/src/github.com/swaggo/swag/cmd/swag/main.go:45 +0x424
```
**To Reproduce**
```
mkdir /tmp/swag
cd /tmp/swag
git init
echo -e "package main\n\nfunc main() {}\n" > main.go
swag init
git add .
git commit -m "test"
git checkout  -b main.go
swag init
```

Workaround:
```
mv .git ../git;swag init;mv ../git .git
```

**Expected behavior**
No panic

**Your swag version**
```
$ swag --version
swag version v1.2.0
```

**Desktop (please complete the following information):**
 - OS: Ubuntu 16.04 LTS

--- a/parser.go
+++ b/parser.go
@@ -619,7 +619,9 @@ func defineTypeOfExample(schemaType string, exampleValue string) interface{} {
 func (parser *Parser) getAllGoFileInfo(searchDir string) {
 	filepath.Walk(searchDir, func(path string, f os.FileInfo, err error) error {
 		//exclude vendor folder
-		if ext := filepath.Ext(path); ext == ".go" && !strings.Contains(string(os.PathSeparator)+path, string(os.PathSeparator)+"vendor"+string(os.PathSeparator)) {
+		if ext := filepath.Ext(path); ext == ".go" &&
+			!strings.Contains(string(os.PathSeparator)+path, string(os.PathSeparator)+"vendor"+string(os.PathSeparator)) &&
+			!strings.Contains(string(os.PathSeparator)+path, string(os.PathSeparator)+".git"+string(os.PathSeparator)) {
 			fset := token.NewFileSet() // positions are relative to fset
 			astFile, err := goparser.ParseFile(fset, path, nil, goparser.ParseComments)
 

